tests/PySys/plugin_apama/apama_plugin_mon_support/run.py

```diff
@@ -0,0 +1,80 @@
+import sys
+
+from environment_apama_plugin import ApamaPlugin
+
+"""
+Validate apama plugin's monitor support.
+Using TedgeTestMonitor mon file in this test's Input directory as the test artefact.
+This project installs and removes this mon file into the apama correlator using the apama plugin.
+"""
+
+
+class ApamaPluginMonSupportTest(ApamaPlugin):
```

Comment:
Took a slightly different approach to test here where all the plugin APIs are tested together in this single test, as opposed to the more unit-test style used for older tests where each plugin API was tested separately. If the former style is preferred, I could break this single test down to smaller individual tests.

Reply:
it looks good, instead of writing multiple tests which duplicate a lot of (setup, cleanup, waiting) code.

Reply:
I definitely prefer this style of end-to-end test.

Reply:
However, this somehow highlights what I don't like with our test framework. It's bizarre to separate the `execute` and `validate` aspects of the code.